### PR TITLE
Create CRUD

### PR DESCRIPTION
**Pull request by thestorozh**

### CODE REVIEW

File: src/main/java/com/storozh/SpringExample/controller/StudentController.java [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    @PostMapping
+    public String create(@RequestBody StudentDTO studentDTO) {
+        studentService.createStudent(studentDTO);
+        LOG.info("Student is created");
+        return "Student is created";
+    }
+
+    @GetMapping
+    public List<Student> getAllStudents() {
+        List<Student> students = studentService.getAllStudents();
```
Comment (quantasync) [inline src/main/java/com/storozh/SpringExample/controller/StudentController.java]: Delete this

File: src/main/java/com/storozh/SpringExample/controller/StudentController.java [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        LOG.info("Retrieved all students");
+        return students;
```
Comment (quantasync) [inline src/main/java/com/storozh/SpringExample/controller/StudentController.java]: return studentService.getAllStudents();

File: src/main/java/com/storozh/SpringExample/DTO/StudentDTO.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.storozh.SpringExample.DTO;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+public class StudentDTO {
+
```
Comment (quantasync) [inline src/main/java/com/storozh/SpringExample/DTO/StudentDTO.java]: middlename? :D

File: src/main/java/com/storozh/SpringExample/DTO/StudentDTO.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.storozh.SpringExample.DTO;
+
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
```
Comment (quantasync) [inline src/main/java/com/storozh/SpringExample/DTO/StudentDTO.java]: Remove @Getter @Setter/ Instead use @Data

File: src/main/java/com/storozh/SpringExample/model/Student.java [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.io.Serializable;
+
+@Getter @Setter
+@Data
+@Entity
+@Table(name = "student")
+public class Student implements Serializable {
+
```
Comment (quantasync) [inline src/main/java/com/storozh/SpringExample/model/Student.java]: middlename?

File: src/main/java/com/storozh/SpringExample/model/Student.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+import lombok.Setter;
+import jakarta.persistence.Column;
+import jakarta.persistence.Entity;
+import jakarta.persistence.GeneratedValue;
+import jakarta.persistence.GenerationType;
+import jakarta.persistence.Id;
+import jakarta.persistence.Table;
+import java.io.Serializable;
+
+@Getter @Setter
```
Comment (quantasync) [inline src/main/java/com/storozh/SpringExample/model/Student.java]: Remove @Getter @Setter

File: src/main/java/com/storozh/SpringExample/service/StudentService.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+            Student student = optionalStudent.get();
+            studentRepository.delete(student);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+	public void deleteStudent(Student student) {
+
```
Comment (quantasync) [inline src/main/java/com/storozh/SpringExample/service/StudentService.java]: Where logic?
Delete this.

File: src/main/java/com/storozh/SpringExample/service/StudentService.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+            return false;
+        }
+    }
+
+	public void deleteStudent(Student student) {
+		
+	}
+
+	public void updateStudent(Student student, StudentDTO studentDTO) {
+
```
Comment (quantasync) [inline src/main/java/com/storozh/SpringExample/service/StudentService.java]: Where logic? It litteraly do nothing :D 
Delete this.

File: src/test/java/com/storozh/SpringExample/SpringExampleApplicationTests.java [hunk 9]
```diff
@@ -0,0 +1,13 @@
+package com.storozh.SpringExample;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class SpringExampleApplicationTests {
```
Comment (quantasync) [inline src/test/java/com/storozh/SpringExample/SpringExampleApplicationTests.java]: Delete this class and package in which it placed.

File: src/main/java/com/storozh/SpringExample/controller/StudentController.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import com.storozh.SpringExample.DTO.StudentDTO;
+import com.storozh.SpringExample.model.Student;
+import java.util.List;
+import com.storozh.SpringExample.service.StudentService;
+
+@RestController
+@RequestMapping(value="/api/student")
```
Comment (SvarmMaster) [inline src/main/java/com/storozh/SpringExample/controller/StudentController.java]: @RequestMapping("/api/student")*

File: src/main/java/com/storozh/SpringExample/controller/StudentController.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+public class StudentController {
+
+    @Autowired
+    StudentService studentService;
+
+    @PostMapping
+    public String create(@RequestBody StudentDTO studentDTO) {
+        studentService.createStudent(studentDTO);
+        return "Student is created";
```
Comment (SvarmMaster) [inline src/main/java/com/storozh/SpringExample/controller/StudentController.java]: return StudentDTO instead string

File: src/main/java/com/storozh/SpringExample/controller/StudentController.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+        return studentService.getAllStudents();
+    }
+
+    @PutMapping("/update/{id}")
+    public ResponseEntity<Student> updateStudent(@PathVariable Long id, @RequestBody StudentDTO studentDTO) {
+        Student student = studentService.updateStudent(id, studentDTO);
+        if (student == null) {
+            return ResponseEntity.notFound().build();
+        }
+        return ResponseEntity.ok(student);
```
Comment (SvarmMaster) [inline src/main/java/com/storozh/SpringExample/controller/StudentController.java]: return StudentDto instead ResponseEntity

File: src/main/java/com/storozh/SpringExample/controller/StudentController.java [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+    @GetMapping
+    public List<Student> getAllStudents() {
+        return studentService.getAllStudents();
+    }
+
+    @PutMapping("/update/{id}")
+    public ResponseEntity<Student> updateStudent(@PathVariable Long id, @RequestBody StudentDTO studentDTO) {
+        Student student = studentService.updateStudent(id, studentDTO);
+        if (student == null) {
+            return ResponseEntity.notFound().build();
```
Comment (SvarmMaster) [inline src/main/java/com/storozh/SpringExample/controller/StudentController.java]: in java 8 you can use Optional to check null value and throw exception. (use RuntimeException)

File: src/main/java/com/storozh/SpringExample/controller/StudentController.java [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+        }
+        return ResponseEntity.ok(student);
+    }
+
+    @DeleteMapping("/delete/{id}")
+    public ResponseEntity<String> deleteStudentById(@PathVariable Long id) {
+        boolean success = studentService.deleteStudentById(id);
+        if (success) {
```
Comment (SvarmMaster) [inline src/main/java/com/storozh/SpringExample/controller/StudentController.java:41]: same

File: src/main/java/com/storozh/SpringExample/service/StudentService.java [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import java.util.List;
+import java.util.Optional;
+
+@Service
+public class StudentService {
+	
+	private static final Logger LOG = LoggerFactory.getLogger(StudentService.class);
```
Comment (SvarmMaster) [inline src/main/java/com/storozh/SpringExample/service/StudentService.java]: align code

File: src/main/java/com/storozh/SpringExample/service/StudentService.java [hunk 16]
```diff
@@ -0,0 +1,60 @@
...
+
+    @Autowired
+    private StudentRepository studentRepository;
+
+    public Student createStudent(StudentDTO studentDTO) {
+        Student student = new Student();
+        student.setFirstname(studentDTO.getFirstname());
+        student.setLastname(studentDTO.getLastname());
+        student.setMiddlename(studentDTO.getMiddlename());
+        studentRepository.save(student);
```
Comment (SvarmMaster) [inline src/main/java/com/storozh/SpringExample/service/StudentService.java]: instead 4 rows above you can use constructor or transformer class.

File: src/main/java/com/storozh/SpringExample/service/StudentService.java [hunk 17]
```diff
@@ -0,0 +1,60 @@
...
+        return student;
+    }
+
+    public List<Student> getAllStudents() {
+        return studentRepository.findAll();
+    }
+
+    public Student updateStudent(Long id, StudentDTO studentDTO) {
+        Optional<Student> optionalStudent = studentRepository.findById(id);
+        if (optionalStudent.isPresent()) {
```
Comment (SvarmMaster) [inline src/main/java/com/storozh/SpringExample/service/StudentService.java:33]: Maybe you can use it.
Student student = optionalStudent.ifPresent(s -> s.toBuilder().firstName(studentDTO.getFirstname())....build());

**Pull request merged**